### DISCUSSION TITLE
Remove address list

### DISCUSSION THREAD

**Discussion by JStika** [tomik23/Leaflet.Autocomplete#26] [Announcements]
Hi,

how remove address list, when I clear input field by keypress backspace?

**Comment by tomickigrzegorz** [comment]
```js
const autocomplete = new Autocomplete("search", {...};
```

```js
document.getElementById("search").addEventListener("keydown", function (e) {
  if (e.key !== "Backspace") return;

  if (e.target.value.length <= 1) {
    autocomplete.destroy();
  }
});

```

PS. You gave me the idea to add another parameter to the library that will clear the results when we do backpspace, whether the input field will be empty 😀 I need to analyze this.

**Comment by JStika** [comment]
Hi,

thank you.
I was first looking for a method that would treat it. 
Something like - [if user  clear input field manualy by "Backspace" or "Del" or other way:
// bool onEvent.InAutoCompleteCondition as: 
// this.inputfield.val() <== this.howManyCharacters
// this is autocomplete

if (onEvent.InAutoCompleteCondition) {
    this.methodNameForRemoveLostOfAddresses();
}

Or

onWatch: (
  if (this.inputfield.val() <== this.howManyCharacters) {
      this.methodNameForRemoveLostOfAddresses();
  }
)
